classes/services/QueryBuilders/BaseQueryBuilder.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/**
+ * @file classes/services/queryBuilders/BaseQueryBuilder.php
+ *
+ * Copyright (c) 2014-2017 Simon Fraser University
+ * Copyright (c) 2000-2017 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class BaseQueryBuilder
+ * @ingroup services_query_builders
+ *
+ * @brief Query builder base class
+ */
+
+namespace App\Services\QueryBuilders;
+
+use Illuminate\Database\Capsule\Manager as Capsule;
+use \Config;
+
+abstract class BaseQueryBuilder {
+
+	/** @var object capsule  */
+	protected $capsule = null;
+
+	/**
+	 * Constructor
+	 */
+	public function __construct() {
+		$this->bootstrap();
+	}
+
+	/**
+	 * bootstrap query builder
+	 */
+	protected function bootstrap() {
+
+		// Map valid OJS3 config options to Illuminate database drivers
+		$driver = strtolower(Config::getVar('database', 'driver'));
+		if (substr($driver, 0, 8) === 'postgres') {
+			$driver = 'pgsql';
+		} else {
+			$driver = 'mysql';
+		}
+
+		// Always use `utf8` unless `latin1` is specified
+		$charset = Config::getVar('i18n', 'connection_charset');
+		if ($charset !== 'latin1') {
+			$charset = 'utf8';
+		}
+
+		$capsule = new Capsule;
+		$capsule->addConnection(array(
```

Comment:
@kaschioudi, here's where I'm wondering whether it's possible to have the QueryBuilder *know* which DBMS it's using without needing to actually connect to it. I don't think the connection is ever used anyway, and introduces a requirement for a properly-configured PDO environment.